fix: npm install from git fails — dist/utils/{logging,yaml} missing

## Problem

`npm install -g github:rysweet/gadugi-agentic-test` installs a broken binary:

```
Error: Cannot find module './logging/LogTransport'
Require stack:
- .../dist/utils/logger.js
```

Then after stubbing that:
```
Error: Cannot find module './yaml/types'
Require stack:
- .../dist/utils/yamlParser.js
```

## Root Cause

**Two issues compound:**

1. **`dist/` is in `.gitignore`** — so `dist/utils/logging/` and `dist/utils/yaml/` (which are generated by `tsc`) are not tracked in git.

2. **Build lifecycle uses `prepublishOnly` instead of `prepare`** — `prepublishOnly` only runs before `npm publish`, NOT before `npm install` from a git URL. The correct hook for git-based installs is `prepare` (runs after git clone + npm install of dependencies).

## Fix

In `package.json`, change:
```json
"prepublishOnly": "npm run build"
```
to:
```json
"prepare": "npm run build"
```

This ensures `tsc` runs after `npm install` clones the repo from GitHub, producing the full `dist/` tree including subdirectories.

## Verification

After cloning and running `npm install && npm run build` manually, the binary works:
```
$ node dist/cli.js --version
1.0.0
```

And the missing files are generated:
```
dist/utils/logging/LogTransport.js  ✓
dist/utils/yaml/types.js            ✓
dist/utils/yaml/YamlLoader.js       ✓
dist/utils/yaml/YamlValidator.js    ✓
dist/utils/yaml/YamlVariableSubstitution.js ✓
```

## Environment
- npm 10.x, Node 22.22.0, Linux
- Install command: `npm install -g github:rysweet/gadugi-agentic-test`

--- a/package.json
+++ b/package.json
@@ -24,10 +24,9 @@
     "lint": "eslint src/**/*.ts",
     "lint:fix": "eslint src/**/*.ts --fix",
     "start": "node dist/cli.js",
-    "prepare": "npm run build",
-    "prepublishOnly": "npm run build",
     "docs": "typedoc && mkdocs serve",
-    "docs:build": "typedoc && mkdocs build --strict"
+    "docs:build": "typedoc && mkdocs build --strict",
+    "prepare": "npm run build"
   },
   "keywords": [
     "testing",

